sim-lib/src/lib.rs

```diff
@@ -292,10 +302,11 @@ impl Simulation {
     async fn generate_activity(
         &self,
         executed_actions: Sender<ActionOutcome>,
-        shutdown: Trigger,
-        listener: Listener,
     ) -> Result<JoinSet<()>, SimulationError> {
         let mut set = JoinSet::new();
+        let shutdown = self.shutdown_trigger.clone();
+        let listener = self.shutdown_listener.clone();
```

Comment:
These aren't inline with `spawn` because of `self` escaping the borrow, right?

Reply:
correct. we can never move `self` into the spawn